rust/src/message/payload/update.rs

```diff
@@ -0,0 +1,432 @@
+    }
+
+    /// Get the update signature field
+    ///
+    /// # Panic
+    ///
+    /// This may panic if the underlying buffer does not represent a
+    /// valid update. If `self.check_buffer_length()` returned
+    /// `Ok(())` this method is guaranteed not to panic.
+    pub fn update_signature(&self) -> &'a [u8] {
+        &self.inner.as_ref()[UPDATE_SIGNATURE_RANGE]
+    }
+}
+
+impl<T: AsRef<[u8]> + AsMut<[u8]>> UpdateBuffer<T> {
+    /// Get a mutable reference to the sum signature field
+    ///
+    /// # Panic
+    ///
+    /// This may panic if the underlying buffer does not represent a
+    /// valid update. If `self.check_buffer_length()` returned
+    /// `Ok(())` this method is guaranteed not to panic.
+    pub fn sum_signature_mut(&mut self) -> &mut [u8] {
+        &mut self.inner.as_mut()[SUM_SIGNATURE_RANGE]
+    }
+
+    /// Get a mutable reference to the update signature field
+    ///
+    /// # Panic
+    ///
+    /// This may panic if the underlying buffer does not represent a
+    /// valid update. If `self.check_buffer_length()` returned
+    /// `Ok(())` this method is guaranteed not to panic.
+    pub fn update_signature_mut(&mut self) -> &mut [u8] {
+        &mut self.inner.as_mut()[UPDATE_SIGNATURE_RANGE]
+    }
+
+    /// Get a mutable reference to the masked model field
+    ///
+    /// # Panic
+    ///
+    /// This may panic if the underlying buffer does not represent a
+    /// valid update. If `self.check_buffer_length()` returned
+    /// `Ok(())` this method is guaranteed not to panic.
+    pub fn masked_model_mut(&mut self) -> LengthValueBuffer<&mut [u8]> {
+        let offset = UPDATE_SIGNATURE_RANGE.end;
+        LengthValueBuffer::new_unchecked(&mut self.inner.as_mut()[offset..])
+    }
+
+    /// Get a mutable reference to the local seed dictionary field
+    ///
+    /// # Panic
+    ///
+    /// This may panic if the underlying buffer does not represent a
+    /// valid update. If `self.check_buffer_length()` returned
+    /// `Ok(())` this method is guaranteed not to panic.
+    pub fn local_seed_dict_mut(&mut self) -> LengthValueBuffer<&mut [u8]> {
+        let offset = UPDATE_SIGNATURE_RANGE.end + self.masked_model_mut().length() as usize;
+        LengthValueBuffer::new_unchecked(&mut self.inner.as_mut()[offset..])
+    }
+}
+
+#[derive(Debug, Eq, PartialEq, Clone)]
+/// High level representation of a update message. These messages are
+/// sent by sum partipants during the update phase.
+pub struct Update<D, M> {
+    /// Signature of the round seed and the word "sum", used to
+    /// determine whether a participant is selected for the sum task
+    pub sum_signature: ParticipantTaskSignature,
+    /// Signature of the round seed and the word "update", used to
+    /// determine whether a participant is selected for the sum task
+    pub update_signature: ParticipantTaskSignature,
+    /// Model trained by an update participant, masked with the
+    /// participant seed
```

Comment:
it's masked with some randomness _derived from_ the seed. so words to that effect would be more accurate